### PR TITLE
Manage datetime for attractivity

### PR DESCRIPTION
**Pull request by azime**
Jira: [NAV-2535](https://navitia.atlassian.net/browse/NAV-2535)

[NAV-2535]: https://navitia.atlassian.net/browse/NAV-2535?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: source/jormungandr/jormungandr/olympic_site_params_manager.py [hunk 1]
```diff
@@ -140,6 +112,14 @@ def get_json_content(self, s3_object):
             logger.exception('Error while loading file: {}'.format(s3_object.key))
             return {}
 
+    def clean_and_get_olympic_site_params(self):
+        result = deepcopy(self.olympic_site_params)
+        for key, value in result.items():
+            for event in value.get("events", []):
+                del event["start"]
+                del event["end"]
+        return result
```
Comment (pbench) [inline source/jormungandr/jormungandr/olympic_site_params_manager.py]: don't make deepcopy, and just return self.olympic_site_params with the "start" and "end" added

File: source/jormungandr/jormungandr/olympic_site_params_manager.py [hunk 2]
```diff
@@ -110,27 +54,55 @@ def build_olympic_site_params(self, scenario, data):
...
+        """
+        transform from_datetime, to_datetime to time_stamp
+        """
+        for key, value in json_data.items():
+            for event in value.get("events", []):
+                str_from_datetime = event.get("from_datetime")
+                str_to_datetime = event.get("to_datetime")
+                if str_from_datetime and str_to_datetime:
+                    event["start"] = self.get_timestamp(str_from_datetime)
+                    event["end"] = self.get_timestamp(str_to_datetime)
```
Comment (pbench) [inline source/jormungandr/jormungandr/olympic_site_params_manager.py]:
```suggestion
                    event["from_datetime_utc"] = self.get_timestamp(str_from_datetime)
                    event["to_datetime_utc"] = self.get_timestamp(str_to_datetime)
```
and rename everywhere else

File: source/jormungandr/jormungandr/interfaces/v1/opg_status.py [hunk 3]
```diff
@@ -39,4 +39,4 @@ def __init__(self, *args, **kwargs):
     def get(self, region=None, lon=None, lat=None):
         region_str = i_manager.get_region(region, lon, lat)
         instance = i_manager.instances[region_str]
-        return instance.olympic_site_params_manager.olympic_site_params, 200
+        return instance.olympic_site_params_manager.clean_and_get_olympic_site_params(), 200
```
Comment (xlqian) [inline source/jormungandr/jormungandr/interfaces/v1/opg_status.py]: that means, the parameters are updated as soon as `/opg_status` is called?

**Review by xlqian — APPROVED** [review]

File: source/jormungandr/jormungandr/olympic_site_params_manager.py (hunk 1 above)
Comment (azime) [reply source/jormungandr/jormungandr/olympic_site_params_manager.py]: Done

File: source/jormungandr/jormungandr/olympic_site_params_manager.py (hunk 2 above)
Comment (azime) [reply source/jormungandr/jormungandr/olympic_site_params_manager.py]: :+1: Done

**Review by xlqian — APPROVED** [review]

**Review by pbench — APPROVED** [review]

**Pull request merged**